lib/server.js

```diff
@@ -80,6 +84,62 @@ class Server {
     });
   }
 
+  startSession(connection, application, strategy, credentials, callback) {
+    const createSession = (username) => {
+      const session = new Session(connection);
+      const sessionStorageProvider = application.sessions || this.sessions;
+      sessionStorageProvider.set(session.id, session);
+      if (sessionStorageProvider.setInactive) {
+        connection.once('close', () => {
+          if (connection.id === session.connection.id) {
+            sessionStorageProvider.setInactive(session.id);
+          }
+        });
+      }
+      callback(null, username, session);
+    };
+
+    if (strategy !== 'anonymous') {
+      if (!this.authPolicy) {
+        callback(errors.ERR_AUTH_FAILED);
+        return;
+      }
+      this.authPolicy(
+        connection, application, strategy, credentials,
+        (error, username) => {
+          if (error || !username) {
+            callback(errors.ERR_AUTH_FAILED);
```

Comment:
I think this has nothing to do with this PR, I didn't change this behavior.
But if you want, you can open another PR with this functionality after this one will be landed.